From a1a831e8677ffd27ce1ef16b375655efa15a5703 Mon Sep 17 00:00:00 2001
From: kenway214 <kenway214@outlook.com>
Date: Sun, 12 Jul 2026 03:05:42 +0530
Subject: [PATCH 14/14] fixup: wm: fix ignore secure window flags

Signed-off-by: kenway214 <kenway214@github.com>
---
 .../server/wm/WindowManagerService.java       | 28 +++++++++++++++++--
 1 file changed, 25 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/wm/WindowManagerService.java b/services/core/java/com/android/server/wm/WindowManagerService.java
index 7bd6cea08947..b2c3f97950f2 100644
--- a/services/core/java/com/android/server/wm/WindowManagerService.java
+++ b/services/core/java/com/android/server/wm/WindowManagerService.java
@@ -119,6 +119,7 @@ import static android.view.displayhash.DisplayHashResultCallback.DISPLAY_HASH_ER
 import static android.view.flags.Flags.sensitiveContentAppProtection;
 import static android.window.ScreenCapture.ScreenCaptureParams.CAPTURE_MODE_REQUIRE_OPTIMIZED;
 import static android.window.ScreenCapture.ScreenCaptureParams.PROTECTED_CONTENT_POLICY_THROW_EXCEPTION;
+import static android.window.ScreenCapture.ScreenCaptureParams.SECURE_CONTENT_POLICY_CAPTURE;
 import static android.window.ScreenCapture.ScreenCaptureParams.SECURE_CONTENT_POLICY_THROW_EXCEPTION;
 import static android.window.WindowProviderService.isWindowProviderService;
 
@@ -863,6 +864,8 @@ public class WindowManagerService extends IWindowManager.Stub
                 Settings.Secure.getUriFor(Settings.Secure.IMMERSIVE_MODE_CONFIRMATIONS);
         private final Uri mDisableSecureWindowsUri =
                 Settings.Secure.getUriFor(Settings.Secure.DISABLE_SECURE_WINDOWS);
+        private final Uri mWindowIgnoreSecureUri =
+                Settings.Global.getUriFor(Settings.Global.WINDOW_IGNORE_SECURE);
         private final Uri mMagnifyImeEnabledUri = Settings.Secure.getUriFor(
                 Settings.Secure.ACCESSIBILITY_MAGNIFICATION_MAGNIFY_NAV_AND_IME);
         private final Uri mPolicyControlUri =
@@ -897,6 +900,8 @@ public class WindowManagerService extends IWindowManager.Stub
                     UserHandle.USER_ALL);
             resolver.registerContentObserver(mDisableSecureWindowsUri, false, this,
                     UserHandle.USER_ALL);
+            resolver.registerContentObserver(mWindowIgnoreSecureUri, false, this,
+                    UserHandle.USER_ALL);
             resolver.registerContentObserver(mMagnifyImeEnabledUri, false, this,
                     UserHandle.USER_ALL);
             resolver.registerContentObserver(mPolicyControlUri, false, this, UserHandle.USER_ALL);
@@ -960,6 +965,11 @@ public class WindowManagerService extends IWindowManager.Stub
                 return;
             }
 
+            if (mWindowIgnoreSecureUri.equals(uri)) {
+                updateWindowIgnoreSecure();
+                return;
+            }
+
             if (mMagnifyImeEnabledUri.equals(uri)) {
                 updateMagnifyIme();
             }
@@ -1098,6 +1108,12 @@ public class WindowManagerService extends IWindowManager.Stub
             }
         }
 
+        void updateWindowIgnoreSecure() {
+            synchronized (mGlobalLock) {
+                mRoot.refreshSecureSurfaceState();
+            }
+        }
+
         void updateMagnifyIme() {
             boolean enabledMagnifyIme = Settings.Secure.getIntForUser(
                     mContext.getContentResolver(),
@@ -11125,10 +11141,16 @@ public class WindowManagerService extends IWindowManager.Stub
             }
         }
 
-        return new ScreenCaptureInternal.LayerCaptureArgs.Builder(
+        final ScreenCaptureInternal.LayerCaptureArgs.Builder builder =
+                new ScreenCaptureInternal.LayerCaptureArgs.Builder(
                         displaySurfaceControl, captureArgs)
-                .setSourceCrop(mTmpRect)
-                .build();
+                        .setSourceCrop(mTmpRect);
+
+        if (getDisableSecureWindows()) {
+            builder.setSecureContentPolicy(SECURE_CONTENT_POLICY_CAPTURE);
+        }
+
+        return builder.build();
     }
 
     @Override
-- 
2.43.0

